### PR TITLE
Feat/authentication authorization

### PR DESCRIPTION
**Pull request by shamir92**
## Summary by CodeRabbit

- **New Features**
  - Introduced new user IT registration functionality.
  - Added bcrypt and JWT management for enhanced security.
  - Added database migration management commands.

- **Improvements**
  - Default server port set to `8080` if not specified.
  - Enhanced database configuration handling and initialization.
  - Updated user entity with NIP validation.

- **Bug Fixes**
  - Fixed SQL script issues with proper termination of statements.

- **Refactor**
  - Renamed package from `commons` to `helper` for better clarity.
  - Updated environment variable names for consistency in database configuration.

- **Documentation**
  - Added documentation for new methods and functionalities.

### CODE REVIEW

**Review by enjaytarigan — APPROVED** [review]
LGTM

File: w3/configuration/bcrypt_configuration.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+func (c *bcryptConfiguration) GetBcryptSalt() int {
+	saltCost, err := strconv.Atoi(c.bcryptSalt)
+
+	if err != nil {
+		log.Fatalf("bcrypt: salt is not a number")
+	}
+
+	return saltCost
```
[the remark this answers is not in the document]
Comment (shamir92) [reply w3/configuration/bcrypt_configuration.go:30]: i force to 16

File: w3/internal/database/postgres_writer.go [hunk 2]
```diff
@@ -4,26 +4,31 @@ import (
...
+	dsn := fmt.Sprintf("postgres://%s:%s@%s:%s/%s?%s",
 		strings.TrimSpace(configDB.GetUser()),
 		strings.TrimSpace(configDB.GetPassword()),
-		strings.TrimSpace(configDB.GetProtocol()),
 		strings.TrimSpace(configDB.GetHost()),
 		strings.TrimSpace(configDB.GetPort()),
 		strings.TrimSpace(configDB.GetName()),
+		strings.TrimSpace(configDB.GetDBParam()),
 	)
+	log.Println(dsn)
```
[the remark this answers is not in the document]
Comment (shamir92) [reply w3/internal/database/postgres_writer.go:31]: done

File: w3/protocol/api/cmd/main.go [hunk 3]
```diff
@@ -15,21 +17,29 @@ func main() {
...
 
 	// For External Interfaces
-	// var postgresWriter postgres.IPostgresWriter = database.NewPostgresWriter(dbConfiguration)
+	postgresWriter, err := database.NewPostgresWriter(dbConfiguration)
+	bcrypt := helper.NewBcryptPasswordHash(bcryptConfiguration)
+	jwtManager := helper.NewJwt(jwtConfiguration)
+
+	if err != nil {
+		log.Fatal(err)
+	}
```
[the remark this answers is not in the document]
Comment (shamir92) [reply w3/protocol/api/cmd/main.go:33]: done set in correct step

File: w3/protocol/api/route/public_route.go [hunk 4]
```diff
@@ -2,35 +2,44 @@ package route
...
 	var pingUsecase usecase.IPingUsecase = usecase.NewPingUsecase()
+	var userITUsecase usecase.IUserITUsecase = usecase.NewUserITUsecase(params.HelperBcrypt, userRepository, params.JWTManager)
 	// TODO: initiation of controller/ handler
 	var pingController controller.IPingController = controller.NewPingController(pingUsecase)
+	var userITController controller.IUserITController = controller.NewUserITController(userITUsecase)
 
 	// Create routes group.
 	route := params.App.Group("/v1")
 	route.Get("/ping", pingController.GetPingController)
+	route.Post("/user/it/register", userITController.RegisterUserIT)
```
[the remark this answers is not in the document]
Comment (shamir92) [reply w3/protocol/api/route/public_route.go:42]: why you remove the line if needed by ping route?
halu ya

**Pull request merged**